lib/puppet/reports/graphite.rb

```diff
@@ -23,11 +26,19 @@ def send_metric payload
 
   def process
     Puppet.debug "Sending status for #{self.host} to Graphite server at #{GRAPHITE_SERVER}"
-    prefix = self.host.split(".").reverse.join(".")
+    
+    if GRAPHITE_REVERSE_HOSTNAME
+      hostname = self.host.split(".").reverse.join(".")      
+    else
+      hostname = self.host
+    end
+    
+    prefix = "#{GRAPHITE_PREFIX}.#{hostname}.#{GRAPHITE_SUFFIX}"
```

Comment:
I believe this will break if the prefix and/or suffix is absent with the dangling . won't it?